requirements.txt

```diff
@@ -1,4 +1,4 @@
-boto3==1.26.54
-kubernetes==12.0.1
-PyYAML==5.4
-pytest-xdist==2.2.0
+boto3==1.28.4
+kubernetes==27.2.0
```

Comment:
As discussed offline, this is a big version bump and we should do a little bit of validation to make sure that we're not going to break everything with this upgrade.  Run some e2e tests locally to lend confidence to this, document in PR description, then I think we're g2g

Reply:
Reduce  the change to only update `PyYAML` to reduce risk. Didn't notice anything break with this upgrade.